webapp-django/crashstats/symbols/views.py

```diff
@@ -29,6 +31,30 @@
 from . import utils
 
 
+metrics = markus.get_metrics('symbols.upload')
+
+
+BAD_CHAR_REGEXP = re.compile(r'[^0-9a-zA-Z\.-]')
+
+
+def _sanitize_email(email):
+    """Gross sanitization of email address so we can use it as a datadog incr tag value
+
+    This converts any non-alphanumeric character to ``_``.
+
+    :arg str email:
+
+    :returns: sanitized email value
+
+    """
+    # First convert from unicode to str dropping any non-ascii characters
+    if isinstance(email, unicode):
+        email = email.encode('ascii', 'ignore')
+
+    # Second, drop any bad characters
+    return BAD_CHAR_REGEXP.sub('', email)
```

Comment:
This is super gross sanitizing code. We really just need to be able to distinguish email addresses from on another. I thought the datadog python client clean up tag keys and values based on what I read, but the code doesn't suggest that at all so I figured we should do it. The `@` gets used to denote tags in the statsd datagrams.

Reply:
This could also use a test. I'll do that tomorrow, too.